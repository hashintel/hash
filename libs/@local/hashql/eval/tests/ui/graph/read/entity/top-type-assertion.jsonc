//@ run: pass
//@ description: Read entity by UUID using scalar property filter
// biome-ignore format: readability
["::core::graph::tail::collect",
  ["::core::graph::body::filter",
    ["::core::graph::head::entities", ["::core::graph::tmp::decision_time_now"]],
    ["fn", { "#tuple": [] }, { "#struct": { "vertex": "_" } }, "_",
      ["is", {"#literal": true}, "Boolean"]
    ]
  ]
]
