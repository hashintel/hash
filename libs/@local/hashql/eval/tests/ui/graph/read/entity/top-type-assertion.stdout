════ HIR ═══════════════════════════════════════════════════════════════════════

::core::graph::head::entities(::core::graph::tmp::decision_time_now())
|> ::core::graph::body::filter(#fn(vertex:0: _0«Entity[(id:
 EntityRecordId[(edition_id: EntityEditionId[Uuid[String]], entity_id:
  EntityId[(draft_id: (Some[DraftId[Uuid[String]]] | None[Null]), entity_uuid:
   EntityUuid[Uuid[String]], web_id:
   WebId[ActorGroupEntityUuid[Uuid[String]]])])], link_data:
 (Some[LinkData[(left_entity_id: EntityId[(draft_id:
    (Some[DraftId[Uuid[String]]] | None[Null]), entity_uuid:
    EntityUuid[Uuid[String]], web_id:
    WebId[ActorGroupEntityUuid[Uuid[String]]])], right_entity_id:
   EntityId[(draft_id: (Some[DraftId[Uuid[String]]] | None[Null]), entity_uuid:
    EntityUuid[Uuid[String]], web_id:
    WebId[ActorGroupEntityUuid[Uuid[String]]])])]]
  | None[Null]), properties: ?)]»): _1«Boolean» ->
    #is(true, type: Boolean)) |> ::core::graph::tail::collect

════ Entity Filter ═════════════════════════════════════════════════════════════

[
    Equal(
        Some(
            Parameter {
                parameter: Boolean(
                    true,
                ),
                convert: None,
            },
        ),
        Some(
            Parameter {
                parameter: Boolean(
                    true,
                ),
                convert: None,
            },
        ),
    ),
]
